frontend/src/app/page.tsx

```diff
@@ -16,15 +17,33 @@ export default function Home() {
     getVids();
   }, []);
 
+  useEffect(() => {
+
+    const getProgress = async () => {
+      if (!videos.length) return
```

Comment:
```suggestion
      if (!videos.length) return

```

Please add a new line after guards, I'm not sure if this is in the linter but it helps readability